### PR TITLE
Orderbook

### PR DESCRIPTION
**Pull request by mkalenska**

### CODE REVIEW

File: lib/arke/orderbook.rb [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+    def initialize(market)
+      @market = market
+      @book = {
+        sell: [],
+        buy: []
+      }
+    end
+
+    def add(order)
+      @book[order.side].push(order) unless @book[order.side].find {|b| b.id == order.id }
```
Comment (dnfd) [inline lib/arke/orderbook.rb]: everything after `unless` is `contains?` method code, just `unless contains?(order)` should work as well

File: lib/arke/orderbook.rb [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+      @book[order.side].delete_at(index) unless index.nil?
+    end
+
+    def update(order)
+      index = @book[order.side].index {|b| b.id == order.id }
+      @book[order.side][index] = order unless index.nil?
+    end
+
+    def contains?(order)
+      @book[order.side].find {|b| b.id == order.id } ? true : false
```
Comment (dnfd) [inline lib/arke/orderbook.rb]: you return true if true, so @book[order.side].find { |b| b.id == order.id } should work as well (and a little space before |b|)

File: spec/arke/orderbook_spec.rb [hunk 3]
```diff
@@ -0,0 +1,118 @@
+require 'spec_helper'
```
Comment (dnfd) [inline spec/arke/orderbook_spec.rb]: spec_helper is required in every spec by rspec

File: lib/arke/orderbook.rb (hunk 2 above)
Comment (mkalenska) [reply lib/arke/orderbook.rb]: book[order.side].find { |b| b.id == order.id } returns nil or found element

File: lib/arke/orderbook.rb [hunk 4]
```diff
@@ -1,10 +1,38 @@
...
+      @book[order.side][index] = order unless index.nil?
+    end
+
+    def contains?(order)
+      @book[order.side].find { |b| b.id == order.id } ? true : false
+    end
+
+    # get with the lowest price
+    def get(side)
+      @book[side].sort_by { |b| b.price }.first
```
Comment (dnfd) [inline lib/arke/orderbook.rb]: As I understand, we have positive and negative prices, anyway, sorting is too expensive here, try witn Array#min, @book[side].min { |a, b| a.price <=> b.price }

**Pull request merged**